x-pack/plugins/endpoint/public/applications/endpoint/components/header_nav.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { MouseEvent } from 'react';
+import { i18n } from '@kbn/i18n';
+import styled from 'styled-components';
+import { EuiTabs, EuiTab } from '@elastic/eui';
+import { useHistory, useLocation } from 'react-router-dom';
+
+export interface NavTabs {
+  name: string;
+  id: string;
+  href: string;
+}
+
+export const navTabs: NavTabs[] = [
+  {
+    id: 'home',
+    name: i18n.translate('xpack.endpoint.headerNav.home', {
+      defaultMessage: 'Home',
+    }),
+    href: '/',
+  },
+  {
+    id: 'management',
+    name: i18n.translate('xpack.endpoint.headerNav.management', {
+      defaultMessage: 'Management',
+    }),
+    href: '/management',
+  },
+  {
+    id: 'alerts',
+    name: i18n.translate('xpack.endpoint.headerNav.alerts', {
+      defaultMessage: 'Alerts',
+    }),
+    href: '/alerts',
+  },
+  {
+    id: 'policies',
+    name: i18n.translate('xpack.endpoint.headerNav.policies', {
+      defaultMessage: 'Policies',
+    }),
+    href: '/policy',
+  },
+];
+
+const Tabs = styled(EuiTabs)`
+  top: 1px;
+  &:before {
+    height: 0px;
+  }
+`;
+
+export const HeaderNavigation: React.FunctionComponent = React.memo(() => {
+  const history = useHistory();
+  const location = useLocation();
+
+  function renderNavTabs(tabs: NavTabs[]) {
+    return tabs.map((tab, index) => {
+      return (
+        <EuiTab
+          data-testid={`${tab.id}EndpointTab`}
+          data-test-subj={`${tab.id}EndpointTab`}
+          key={index}
+          href={`/app/endpoint${tab.href}`}
+          onClick={(event: MouseEvent) => {
```

Comment:
An idea for how we could emulate react-router when using Eui components.
https://github.com/oatkiller/kibana/commit/d6861d858a432f43d2fa48792beaa79f3ef43e23

Things this handles:
1. useCallback for the callback
2. doesn't navigate when modifier keys are pressed
3. can call an additional onClick handler if needed (navigate via a link AND have a callback)
4. doesn't navigate if event is 'defaultPrevented'
5. logic copied from react-router's link, so behavior is consistent
6. should work in EuiLink as well

see react router's implementation for ideas
https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/modules/Link.js

Reply:
Great approach that prevents us having to wrap Eui components 👍

Reply:
@oatkiller @paul-tavares this doesn't quite work yet, I broke this out to another ticket: https://github.com/elastic/endpoint-app-team/issues/230

Are you OK to merge this as is and I'll investigate the next step for this?